### PR TITLE
Rewards Documentation

### PR DESCRIPTION
**Pull request by beaushinkle**
This PR adds documentation for the rewards mechanism of the sortition pool!

Tagging @lukasz-zimnoch @eth-r @michalinacienciala and @pdyraga for review

closes https://github.com/keep-network/sortition-pools/issues/159

### CODE REVIEW

**Review by pdyraga — APPROVED** [review]

**Comment by pdyraga** [comment]
We could also mention [`restoreEligibility`](https://github.com/keep-network/sortition-pools/blob/f937be7f4e23789b045e02e96a0d0682c8e20bee/contracts/Rewards.sol#L190) and how it's updating the state.

File: docs/rewards.md [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+snapshot of the `globalRewardAccumulator` at the time they were last updated.
+Their `available` value represents how much reward is available for withdraw.
+Their `weight` is their weight in the pool.
+
+To see how all of these pieces of state interact, we can go through some
+event logs.
+
+#### Join -> Reward -> Withdraw
+```
+event 1) Alice (id 0) joins the pool with weight 10
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rewards.md:47]: good call

File: docs/rewards.md [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+
+The (1) `globalRewardAccumulator` represents how much reward a 1-`weight`
+operator would have accumulated since genesis. Since we're working in integers,
+and since rewards won't always be cleanly divisible by the pool weight, there
+carry-over, which is stored in (2) `rewardRoundingDust`.
+
+Finally, the (3) `operatorRewards` keep track of each operator's individual
+state indexed by their `id`. An operator's `accumulated` value represents a
+snapshot of the `globalRewardAccumulator` at the time they were last updated.
+Their `available` value represents how much reward is available for withdraw.
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rewards.md:39]: [8cfb027](https://github.com/keep-network/sortition-pools/pull/180/commits/8cfb02771d2cedd0b3482a2802e9c89128a5ed99)

File: docs/rewards.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rewards.md:47]: [12bd114](https://github.com/keep-network/sortition-pools/pull/180/commits/12bd11417ebac297ab81072232ec9fe60e5c6d6b)

**Pull request merged**